x-pack/plugins/fleet/server/services/security/uninstall_token_service/index.ts

```diff
@@ -131,52 +178,48 @@ export class UninstallTokenService implements UninstallTokenServiceInterface {
       break;
     }
 
-    filter = aggResults
-      .reduce((acc, { latest }) => {
-        const id = latest?.hits?.hits?.at(0)?._id;
-        if (!id) return acc;
-        const filterStr = `${UNINSTALL_TOKENS_SAVED_OBJECT_TYPE}.id: "${id}"`;
-        acc.push(filterStr);
-        return acc;
-      }, [] as string[])
+    if (aggResults.length === 0) {
+      return { items: [], total: 0, page, perPage };
+    }
+
+    const filter: string = aggResults
+      .filter(({ latest }) => latest?.hits?.hits?.at(0)?._id)
+      .map(({ latest }) => {
+        return `${UNINSTALL_TOKENS_SAVED_OBJECT_TYPE}.id: "${latest?.hits?.hits?.at(0)?._id}"`;
+      })
       .join(' or ');
 
     const tokensFinder =
       await this.esoClient.createPointInTimeFinderDecryptedAsInternalUser<UninstallTokenSOAttributes>(
         {
           type: UNINSTALL_TOKENS_SAVED_OBJECT_TYPE,
-          perPage: SO_SEARCH_LIMIT,
+          perPage,
           filter,
         }
       );
     let tokenObjects: Array<SavedObjectsFindResult<UninstallTokenSOAttributes>> = [];
+
+    let currentPage = 1;
```

Comment:
Personally, I would avoid doing extra ES calls for pagination but would also know others thoughts.

Reply:
yep, n queries for the nth page. agree to avoid this, here's a solution I tried to come up with: 5c7db44

to sum up the algo:
- first there is a terms aggregation for finding the latest tokens (without decryption, because agg and decrypt doesn't go along) -> this query cannot really be sorted due to the terms agg  ([docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html#_ordering_by_count_ascending))
- `[new]` then the received buckets are sorted in JS world by `created_at: 'desc'`
- `[new]` then the SO IDs for the current page are sliced from the sorted array
- and at last we have the decrypt query which simply filters using the IDs (because it doesn't support neither pagination nor terms aggregation out of the box)

so we traded some ES queries for a `O(n*log(n))` sorting, which I hope is a win :)

what do you think?